test_framework/testdata.py

```diff
@@ -45,15 +45,18 @@ def read(name):
 
 
     # Resolve links
+    _data = {}
     for k in data:
```

Comment:
Couldn't we use `list()` here to make a copy of the keys?
```suggestion
    for k in list(data):
```